custom/enikshay/integrations/nikshay/repeater_generator.py

```diff
@@ -327,6 +338,93 @@ def handle_exception(self, exception, repeat_record):
                                 "followup_nikshay_registered", "followup_nikshay_error")
 
 
+class NikshayRegisterPrivatePatientPayloadGenerator(BaseNikshayPayloadGenerator):
+    format_name = 'case_xml'
+    format_label = 'XML'
+
+    @memoized
```

Comment:
@dannyroberts any thoughts? @proteusvacuum 
From the payload that was sent in request, i need a field(HFIDNO) from it in `handle_success` of the corresponding repeater.
So i have called `repeat_record.get_payload` in handle_success, which eventually calls this and memoized it assuming that it would all happen in one request.
**Pros**: Computation/SQL-Queries skipped, same value as sent was used just in case recomputing gives a different value though highly unlikely

Does it look a good idea? Or it would be harmful?
Should it be done on the parent class?

Other way would be to recompute the value needed or pass the payload as an argument which would need me to update all handle_success.